pkg/api/kptfile/v1alpha2/validation.go

```diff
@@ -147,13 +147,21 @@ func IsNodeZero(n *yaml.Node) bool {
 		n.Line == 0 && n.Column == 0
 }
 
-// validatePath validates input path and return an error if it's invalid
-func validatePath(p string) error {
+// validateFnConfigPath validates given function config path and return an error if it's invalid
+func validateFnConfigPath(p string) error {
+	if strings.TrimSpace(p) == "" {
+		return fmt.Errorf("path must not be empty")
+	}
+	p = path.Clean(p)
 	if path.IsAbs(p) {
-		return fmt.Errorf("path is not relative")
+		return fmt.Errorf("path must be relative")
 	}
-	if strings.TrimSpace(p) == "" {
-		return fmt.Errorf("path cannot have only white spaces")
+	if strings.HasPrefix(p, "..") {
```

Comment:
HasPrefix -> Contains would be a stronger check

Reply:
similar to the concern here https://github.com/GoogleContainerTools/kpt/pull/2027#discussion_r633937293 is just checking for ".." enough? Should we check for HasPrefix(path.Clean(p)) to account for cases like `a/../../b`?

Reply:
Using `strings.Contains` now.

@natasha41575 path is already cleaned above. So contains will catch `../b` that you described. Also added more unit test cases to test this case.